### PR TITLE
[NEW RESOURCE] AWS Backup Restore Testing

### PR DESCRIPTION
**Pull request by Thomas-Franklin**
### Description

Introduces the `aws_backup_restore_testing_plan` and `aws_backup-restore_selection` resources. These resources manages the AWS Backup Restore Testing functionality within AWS.

### Relations

Closes https://github.com/hashicorp/terraform-provider-aws/issues/34699.
Closes https://github.com/hashicorp/terraform-provider-aws/pull/36907.

### References

* https://docs.aws.amazon.com/cli/latest/reference/backup/create-restore-testing-selection.html
* https://docs.aws.amazon.com/cli/latest/reference/backup/create-restore-testing-plan.html
* https://aws.amazon.com/blogs/aws/automatic-restore-testing-and-validation-is-now-available-in-aws-backup/

### Output from Acceptance Testing

BackupRestoreTestingPlan:

```console
make testacc TESTS=TestAccBackupRestoreTestingPlan_\* PKG=backup
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/backup/... -v -count 1 -parallel 20 -run='TestAccBackupRestoreTestingPlan_*'  -timeout 360m
=== RUN   TestAccBackupRestoreTestingPlan_basic
=== PAUSE TestAccBackupRestoreTestingPlan_basic
=== RUN   TestAccBackupRestoreTestingPlan_disappears
=== PAUSE TestAccBackupRestoreTestingPlan_disappears
=== RUN   TestAccBackupRestoreTestingPlan_tags
=== PAUSE TestAccBackupRestoreTestingPlan_tags
=== RUN   TestAccBackupRestoreTestingPlan_includevaults
=== PAUSE TestAccBackupRestoreTestingPlan_includevaults
=== RUN   TestAccBackupRestoreTestingPlan_excludevaults
=== PAUSE TestAccBackupRestoreTestingPlan_excludevaults
=== RUN   TestAccBackupRestoreTestingPlan_additionals
=== PAUSE TestAccBackupRestoreTestingPlan_additionals
=== RUN   TestAccBackupRestoreTestingPlan_additionalwithupdates
=== PAUSE TestAccBackupRestoreTestingPlan_additionalwithupdates
=== CONT  TestAccBackupRestoreTestingPlan_basic
=== CONT  TestAccBackupRestoreTestingPlan_excludevaults
=== CONT  TestAccBackupRestoreTestingPlan_tags
=== CONT  TestAccBackupRestoreTestingPlan_additionalwithupdates
=== CONT  TestAccBackupRestoreTestingPlan_additionals
=== CONT  TestAccBackupRestoreTestingPlan_includevaults
=== CONT  TestAccBackupRestoreTestingPlan_disappears
--- PASS: TestAccBackupRestoreTestingPlan_disappears (11.29s)
--- PASS: TestAccBackupRestoreTestingPlan_basic (13.65s)
--- PASS: TestAccBackupRestoreTestingPlan_additionals (13.67s)
--- PASS: TestAccBackupRestoreTestingPlan_includevaults (19.38s)
--- PASS: TestAccBackupRestoreTestingPlan_excludevaults (19.41s)
--- PASS: TestAccBackupRestoreTestingPlan_tags (20.19s)
--- PASS: TestAccBackupRestoreTestingPlan_additionalwithupdates (21.60s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/backup	25.337s
```

BackupRestoreTestingSelection:

```console
make testacc TESTS=TestAccBackupRestoreTestingSelection_\* PKG=backup
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/backup/... -v -count 1 -parallel 20 -run='TestAccBackupRestoreTestingSelection_*'  -timeout 360m
=== RUN   TestAccBackupRestoreTestingSelection_basic
=== PAUSE TestAccBackupRestoreTestingSelection_basic
=== RUN   TestAccBackupRestoreTestingSelection_disappears
=== PAUSE TestAccBackupRestoreTestingSelection_disappears
=== RUN   TestAccBackupRestoreTestingSelection_updates
=== PAUSE TestAccBackupRestoreTestingSelection_updates
=== CONT  TestAccBackupRestoreTestingSelection_basic
=== CONT  TestAccBackupRestoreTestingSelection_disappears
=== CONT  TestAccBackupRestoreTestingSelection_updates
--- PASS: TestAccBackupRestoreTestingSelection_disappears (17.21s)
--- PASS: TestAccBackupRestoreTestingSelection_basic (20.33s)
--- PASS: TestAccBackupRestoreTestingSelection_updates (29.72s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/backup	33.571s
```

### CODE REVIEW

**Comment by mervin-hemaraju-cko** [comment]
Heyy, when is this getting merged?

**Comment by Nirrleybo** [comment]
Any update on this being merged in?

**Comment by Thomas-Franklin** [comment]
> Any update on this being merged in?

Not heard anything, but I have rebased it to make it easier for the owners to merge.

**Comment by ricoleabricot** [comment]
Up-ing this PR, as I think it could be merged as it (and recently rebased).
@justinretzolk any way we could have an ETA of merge + next provider version?

File: internal/service/backup/find.go [hunk 1]
```diff
@@ -6,9 +6,9 @@ package backup
 import (
 	"context"
 
-	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/service/backup"
 	awstypes "github.com/aws/aws-sdk-go-v2/service/backup/types"
+	"github.com/aws/aws-sdk-go/aws"
```
Comment (aristosvo) [inline internal/service/backup/find.go:11]: I think there are some changes not rebased entirely correct, like this one:
```suggestion
	"github.com/aws/aws-sdk-go-v2/aws"
```

File: internal/service/backup/find.go [hunk 2]
```diff
@@ -136,3 +136,52 @@ func findFrameworkByName(ctx context.Context, conn *backup.Client, name string)
 
 	return output, nil
 }
+
+func FindRestoreTestingPlanByName(ctx context.Context, conn *backup.Client, name string) (*backup.GetRestoreTestingPlanOutput, error) {
```
Comment (aristosvo) [inline internal/service/backup/find.go:140]: Lowercase/private seems to be the standard for now
```suggestion
func findRestoreTestingPlanByName(ctx context.Context, conn *backup.Client, name string) (*backup.GetRestoreTestingPlanOutput, error) {
```

File: internal/service/backup/find.go [hunk 3]
```diff
@@ -136,3 +136,52 @@ func findFrameworkByName(ctx context.Context, conn *backup.Client, name string)
...
+	}
+
+	if out == nil || out.RestoreTestingPlan == nil {
+		return nil, tfresource.NewEmptyResultError(in)
+	}
+
+	return out, nil
+}
+
+func FindRestoreTestingSelectionByName(ctx context.Context, conn *backup.Client, name string, restoreTestingPlanName string) (*backup.GetRestoreTestingSelectionOutput, error) {
```
Comment (aristosvo) [inline internal/service/backup/find.go:164]: Same here
```suggestion
func findRestoreTestingSelectionByName(ctx context.Context, conn *backup.Client, name string, restoreTestingPlanName string) (*backup.GetRestoreTestingSelectionOutput, error) {
```

File: internal/service/backup/restore_testing_plan.go [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/YakDriver/regexache"
+	"github.com/aws/aws-sdk-go-v2/service/backup"
+	awstypes "github.com/aws/aws-sdk-go-v2/service/backup/types"
+	"github.com/aws/aws-sdk-go/aws"
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan.go:14]: v2 again :)
```suggestion
	"github.com/aws/aws-sdk-go-v2/aws"
```

File: internal/service/backup/restore_testing_plan.go [hunk 5]
```diff
@@ -0,0 +1,461 @@
...
+	framework.ResourceWithConfigure
+	framework.WithImportByID
+	framework.WithTimeouts
+}
+
+func (r *restoreTestingPlanResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_backup_restore_testing_plan"
+}
+
+// ==== SCHEMA ==== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan.go:64]: Clean :)
```suggestion
```

File: internal/service/backup/restore_testing_plan.go [hunk 6]
```diff
@@ -0,0 +1,461 @@
...
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root(names.AttrName), req.ID)...)
+}
+
+// ==== STATE REFRESH ==== //
+const (
+	stateNormal   = "NORMAL"
+	stateNotFound = "NOT_FOUND"
+)
+
+// ==== FUNCTIONS ==== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan.go:392]: I think these aren't necessary
```suggestion
```

File: internal/service/backup/restore_testing_plan.go [hunk 7]
```diff
@@ -0,0 +1,461 @@
...
+
+func (r *restoreTestingPlanResource) ModifyPlan(ctx context.Context, request resource.ModifyPlanRequest, response *resource.ModifyPlanResponse) {
+	r.SetTagsAll(ctx, request, response)
+}
+
+func (r *restoreTestingPlanResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root(names.AttrName), req.ID)...)
+}
+
+// ==== STATE REFRESH ==== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan.go:386]: Unnecessary comment I think?
```suggestion
```

File: internal/service/backup/restore_testing_plan.go [hunk 8]
```diff
@@ -0,0 +1,461 @@
...
+		Timeout: timeout,
+	}
+	outputRaw, err := stateConf.WaitForStateContext(ctx)
+
+	if out, ok := outputRaw.(*backup.GetRestoreTestingPlanOutput); ok {
+		return out, err
+	}
+
+	return nil, err
+
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan.go:407]:
```suggestion
```

File: internal/service/backup/restore_testing_plan.go [hunk 9]
```diff
@@ -0,0 +1,461 @@
...
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan.go:424]:
```suggestion
```

File: internal/service/backup/restore_testing_plan.go [hunk 10]
```diff
@@ -0,0 +1,461 @@
...
+		if err != nil {
+			return nil, "", err
+		}
+
+		return output, stateNormal, nil
+	}
+}
+
+// ==== DATA STRUCTURES ==== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan.go:442]:
```suggestion
```

File: internal/service/backup/restore_testing_plan_test.go [hunk 11]
```diff
@@ -0,0 +1,501 @@
...
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
+	"github.com/hashicorp/terraform-provider-aws/names"
+
+	tfbackup "github.com/hashicorp/terraform-provider-aws/internal/service/backup"
+)
+
+// ==== ACCEPTANCE TESTS ==== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan_test.go:27]:
```suggestion
```

File: internal/service/backup/restore_testing_plan_test.go [hunk 12]
```diff
@@ -0,0 +1,501 @@
...
+				Config: testAccRestoreTestingPlanConfig_tags("Name", "RestoreTestingPlan", rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRestoreTestingPlanExists(ctx, resourceName, &restoretestingplan),
+					resource.TestCheckResourceAttrSet(resourceName, "arn"),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "recovery_point_selection.0.algorithm", "LATEST_WITHIN_WINDOW"),
+					resource.TestCheckResourceAttr(resourceName, "recovery_point_selection.0.include_vaults.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "recovery_point_selection.0.recovery_point_types.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "schedule_expression", "cron(0 12 ? * * *)"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"), // has tags
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan_test.go:117]:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
```

File: internal/service/backup/restore_testing_plan_test.go [hunk 13]
```diff
@@ -0,0 +1,501 @@
...
+				Config: testAccRestoreTestingPlanConfig_tags("Name", "Testing1", rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRestoreTestingPlanExists(ctx, resourceName, &restoretestingplan),
+					resource.TestCheckResourceAttrSet(resourceName, "arn"),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "recovery_point_selection.0.algorithm", "LATEST_WITHIN_WINDOW"),
+					resource.TestCheckResourceAttr(resourceName, "recovery_point_selection.0.include_vaults.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "recovery_point_selection.0.recovery_point_types.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "schedule_expression", "cron(0 12 ? * * *)"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"), // has tags
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan_test.go:139]:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
```

File: internal/service/backup/restore_testing_plan_test.go [hunk 14]
```diff
@@ -0,0 +1,501 @@
...
+					resource.TestCheckResourceAttr(resourceName, "tags.Name", "Testing1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccBackupRestoreTestingPlan_includevaults(t *testing.T) {
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan_test.go:147]:
```suggestion
func TestAccBackupRestoreTestingPlan_includeVaults(t *testing.T) {
```

File: internal/service/backup/restore_testing_plan_test.go [hunk 15]
```diff
@@ -0,0 +1,501 @@
...
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			testAccPreCheck(ctx, t)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, names.BackupServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckRestoreTestingPlanDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRestoreTestingPlanConfig_includevaults(rName),
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan_test.go:162]:
```suggestion
				Config: testAccRestoreTestingPlanConfig_includeVaults(rName),
```

File: internal/service/backup/restore_testing_plan_test.go [hunk 16]
```diff
@@ -0,0 +1,501 @@
...
+				ImportStateVerify:                    true,
+				ImportStateId:                        rName,
+				ImportStateVerifyIdentifierAttribute: "name",
+				ImportStateVerifyIgnore:              []string{"apply_immediately", "user"},
+			},
+		},
+	})
+}
+
+func TestAccBackupRestoreTestingPlan_excludevaults(t *testing.T) {
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan_test.go:186]:
```suggestion
func TestAccBackupRestoreTestingPlan_excludeVaults(t *testing.T) {
```

File: internal/service/backup/restore_testing_plan_test.go [hunk 17]
```diff
@@ -0,0 +1,501 @@
...
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			testAccPreCheck(ctx, t)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, names.BackupServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckRestoreTestingPlanDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRestoreTestingPlanConfig_excludevaults(rName),
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan_test.go:201]:
```suggestion
				Config: testAccRestoreTestingPlanConfig_excludeVaults(rName),
```

File: internal/service/backup/restore_testing_plan_test.go [hunk 18]
```diff
@@ -0,0 +1,501 @@
...
+					resource.TestCheckResourceAttr(resourceName, "schedule_expression_timezone", "Europe/London"),
+					resource.TestCheckResourceAttr(resourceName, "start_window_hours", "12"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Name", "RestoreTestingPlan"),
+				),
+			},
+		},
+	})
+}
+
+// ====== Helper Functions ====== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan_test.go:344]:
```suggestion
```

File: internal/service/backup/restore_testing_plan_test.go [hunk 19]
```diff
@@ -0,0 +1,501 @@
...
+			return create.Error(names.Backup, create.ErrActionCheckingExistence, tfbackup.ResNameRestoreTestingPlan, name, err)
+		}
+
+		*restoretestingplan = *resp
+
+		return nil
+	}
+}
+
+// ====== Terraform Configuration Functions ====== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan_test.go:396]:
```suggestion
```

File: internal/service/backup/restore_testing_plan_test.go [hunk 20]
```diff
@@ -0,0 +1,501 @@
...
+		  selection_window_days = %[1]s
+		}
+  
+		schedule_expression = "%[2]s"
+		start_window_hours = 168
+  	}
+`, selectionWindowDays, scheduleExpression, rName)
+}
+
+func testAccRestoreTestingPlanConfig_includevaults(rName string) string {
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan_test.go:464]:
```suggestion
func testAccRestoreTestingPlanConfig_includeVaults(rName string) string {
```

File: internal/service/backup/restore_testing_plan_test.go [hunk 21]
```diff
@@ -0,0 +1,501 @@
...
+		  recovery_point_types = ["CONTINUOUS"]
+		}
+  
+		schedule_expression = "cron(0 12 ? * * *)" # Daily at 12:00
+  	}
+`, rName),
+	)
+}
+
+func testAccRestoreTestingPlanConfig_excludevaults(rName string) string {
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_plan_test.go:483]:
```suggestion
func testAccRestoreTestingPlanConfig_excludeVaults(rName string) string {
```

File: internal/service/backup/restore_testing_selection.go [hunk 22]
```diff
@@ -0,0 +1,421 @@
...
+type resourceRestoreTestingSelection struct {
+	framework.ResourceWithConfigure
+	framework.WithTimeouts
+}
+
+func (r *resourceRestoreTestingSelection) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_backup_restore_testing_selection"
+}
+
+// ==== SCHEMA ==== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_selection.go:65]:
```suggestion
```

File: internal/service/backup/restore_testing_selection.go [hunk 23]
```diff
@@ -0,0 +1,421 @@
...
+
+		if err != nil {
+			return nil, "", err
+		}
+
+		return output, stateNormal, nil
+	}
+}
+
+// ==== DATA STRUCTURES ==== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_selection.go:401]:
```suggestion
```

File: internal/service/backup/restore_testing_selection_test.go [hunk 24]
```diff
@@ -0,0 +1,279 @@
...
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
+	tfbackup "github.com/hashicorp/terraform-provider-aws/internal/service/backup"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// ==== ACCEPTANCE TESTS ==== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_selection_test.go:26]:
```suggestion
```

File: internal/service/backup/restore_testing_selection_test.go [hunk 25]
```diff
@@ -0,0 +1,279 @@
...
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "restore_testing_plan_name", rName),
+					resource.TestCheckResourceAttr(resourceName, "protected_resource_type", "EC2"),
+				),
+			},
+		},
+	})
+}
+
+// ====== Helper Functions ====== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_selection_test.go:132]:
```suggestion
```

File: internal/service/backup/restore_testing_selection_test.go [hunk 26]
```diff
@@ -0,0 +1,279 @@
...
+			return create.Error(names.Backup, create.ErrActionCheckingExistence, tfbackup.ResNameRestoreTestingPlan, name, err)
+		}
+
+		*restoretestingplan = *resp
+
+		return nil
+	}
+}
+
+// ====== Terraform Configuration Functions ====== //
```
Comment (aristosvo) [inline internal/service/backup/restore_testing_selection_test.go:184]:
```suggestion
```

File: names/data/names_data.csv
Comment (aristosvo) [inline names/data/names_data.csv:1]: Please delete, this is due to the rebase out of date

**Review by aristosvo — CHANGES_REQUESTED** [review]
Hi @Thomas-Franklin!

Thanks for this PR! I'm 'just' another contributor who is also eager to get this resource in (better today than tomorrow 😄 ). I went through your PR and flagged some stuff that could use some change on a first look. I'm not a maintainer and cannot push this PR through, so don't expect too much of it 😺 

If my comments are a bit short and need some explanation, please let me know.

**Comment by Thomas-Franklin** [comment]
@aristosvo - I believe I have addressed all your suggestions, appreciate the kind review!

File: internal/service/backup/exports_test.go [hunk 27]
```diff
@@ -5,6 +5,10 @@ package backup
 
 // Exports for use in tests only.
 var (
-	FindVaultAccessPolicyByName = findVaultAccessPolicyByName
-	FindVaultByName             = findVaultByName
+	FindVaultAccessPolicyByName       = findVaultAccessPolicyByName
+	FindVaultByName                   = findVaultByName
+	FindRestoreTestingPlanByName      = findRestoreTestingPlanByName
+	findRestoreTestingSelectionByName = findRestoreTestingSelectionByName
```
Comment (aristosvo) [inline internal/service/backup/exports_test.go:11]:
```suggestion
	FindRestoreTestingSelectionByName = findRestoreTestingSelectionByName
```

**Comment by aristosvo** [comment]
@Thomas-Franklin I asked the HC maintainers to enable all linters/checks, some work to do😅

**Comment by Thomas-Franklin** [comment]
> @Thomas-Franklin I asked the HC maintainers to enable all linters/checks, some work to do😅

I will take a look tomorrow at the failing actions, thanks for all the support 🙏

**Comment by Thomas-Franklin** [comment]
Okay I did run the lint fix @aristosvo but it changed 4000 markdown files... I suppose I should commit just my markdown files?

**Comment by aristosvo** [comment]
> Okay I did run the lint fix @aristosvo but it changed 4000 markdown files... I suppose I should commit just my markdown files?

Yes, definitely. Besides that, I noticed you changed a lot in `names/data/names_data.hcl`, that might be the cause of some of these issues as well.

**Comment by Thomas-Franklin** [comment]
Hi @justinretzolk - rebased and solved conflicts again, with a clean up of suggestions by @aristosvo ; just wondering what we can do to get this further along in for a release. 🤞

**Comment by mervin-hemaraju-cko** [comment]
Hello all, could we please take a look and merge this? 🙌🏽

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccBackupRestoreTesting' PKG=backup ACCTEST_PARALLELISM=3 
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.1 test ./internal/service/backup/... -v -count 1 -parallel 3  -run=TestAccBackupRestoreTesting -timeout 360m
=== RUN   TestAccBackupRestoreTestingPlan_basic
=== PAUSE TestAccBackupRestoreTestingPlan_basic
=== RUN   TestAccBackupRestoreTestingPlan_disappears
=== PAUSE TestAccBackupRestoreTestingPlan_disappears
=== RUN   TestAccBackupRestoreTestingPlan_tags
=== PAUSE TestAccBackupRestoreTestingPlan_tags
=== RUN   TestAccBackupRestoreTestingPlan_includeVaults
=== PAUSE TestAccBackupRestoreTestingPlan_includeVaults
=== RUN   TestAccBackupRestoreTestingPlan_excludeVaults
=== PAUSE TestAccBackupRestoreTestingPlan_excludeVaults
=== RUN   TestAccBackupRestoreTestingPlan_additionals
=== PAUSE TestAccBackupRestoreTestingPlan_additionals
=== RUN   TestAccBackupRestoreTestingPlan_additionalsWithUpdate
=== PAUSE TestAccBackupRestoreTestingPlan_additionalsWithUpdate
=== RUN   TestAccBackupRestoreTestingSelection_basic
=== PAUSE TestAccBackupRestoreTestingSelection_basic
=== RUN   TestAccBackupRestoreTestingSelection_disappears
=== PAUSE TestAccBackupRestoreTestingSelection_disappears
=== RUN   TestAccBackupRestoreTestingSelection_updates
=== PAUSE TestAccBackupRestoreTestingSelection_updates
=== CONT  TestAccBackupRestoreTestingPlan_basic
=== CONT  TestAccBackupRestoreTestingPlan_additionals
=== CONT  TestAccBackupRestoreTestingPlan_includeVaults
--- PASS: TestAccBackupRestoreTestingPlan_basic (13.85s)
=== CONT  TestAccBackupRestoreTestingPlan_excludeVaults
--- PASS: TestAccBackupRestoreTestingPlan_additionals (13.86s)
=== CONT  TestAccBackupRestoreTestingPlan_tags
--- PASS: TestAccBackupRestoreTestingPlan_includeVaults (20.99s)
=== CONT  TestAccBackupRestoreTestingPlan_disappears
--- PASS: TestAccBackupRestoreTestingPlan_disappears (10.45s)
=== CONT  TestAccBackupRestoreTestingSelection_disappears
--- PASS: TestAccBackupRestoreTestingPlan_excludeVaults (19.68s)
=== CONT  TestAccBackupRestoreTestingSelection_updates
--- PASS: TestAccBackupRestoreTestingPlan_tags (27.97s)
=== CONT  TestAccBackupRestoreTestingPlan_additionalsWithUpdate
--- PASS: TestAccBackupRestoreTestingSelection_disappears (16.80s)
=== CONT  TestAccBackupRestoreTestingSelection_basic
--- PASS: TestAccBackupRestoreTestingSelection_updates (26.94s)
--- PASS: TestAccBackupRestoreTestingPlan_additionalsWithUpdate (20.10s)
--- PASS: TestAccBackupRestoreTestingSelection_basic (18.62s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/backup	72.175s
```

**Comment by jeroenhabets** [comment]
@ewbankkit as [per your comment](https://github.com/hashicorp/terraform-provider-aws/pull/37039#pullrequestreview-2348865278) this PR seems to be about to get merged, could you please close competing PR https://github.com/hashicorp/terraform-provider-aws/pull/36907? TIA!

**Pull request merged**